File: ios/KSTFunMetro/NativeTextInputManager.m
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(NativeTextInputManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(value, NSString)
RCT_EXPORT_VIEW_PROPERTY(placeholder, NSString)
RCT_EXPORT_VIEW_PROPERTY(secureTextEntry, BOOL)
RCT_EXPORT_VIEW_PROPERTY(onChangeText, RCTDirectEventBlock)

@end

File: ios/KSTFunMetro/NativeTextInputManager.swift
import UIKit
import React

@objc(NativeTextInputManager)
class NativeTextInputManager: RCTViewManager {
  
  override func view() -> UIView! {
    return NativeTextInputView()
  }
  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  // Export props to React Native
  @objc func setValue(_ node: NSNumber, value: String) {
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view(forReactTag: node) as! NativeTextInputView
      component.setValue(value)
    }
  }
  
  @objc func setPlaceholder(_ node: NSNumber, placeholder: String) {
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view(forReactTag: node) as! NativeTextInputView
      component.setPlaceholder(placeholder)
    }
  }
  
  @objc func setSecureTextEntry(_ node: NSNumber, secure: Bool) {
    DispatchQueue.main.async {
      let component = self.bridge.uiManager.view(forReactTag: node) as! NativeTextInputView
      component.setSecureTextEntry(secure)
    }
  }
}